clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -533,6 +533,12 @@ record_filter_t::process_markers(per_shard_t *per_shard, trace_entry_t &entry,
                        "supported";
             }
             break;
+        case TRACE_MARKER_TYPE_CORE_WAIT:
+            // These are artificial timing records: do not output them, nor consider
+            // them real input records.
+            output = false;
+            --per_shard->input_entry_count;
```

Comment:
Could we generalize this using is_record_synthetic?

Reply:
No since that includes IDLE and we do want to include IDLE for core-sharded to capture empty periods.